onedal/common/sycl.cpp

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+* Copyright 2024 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "onedal/common/sycl_interfaces.hpp"
+#include "onedal/common/pybind11_helpers.hpp"
+
+namespace py = pybind11;
+
+namespace oneapi::dal::python {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+void instantiate_sycl_interfaces(py::module& m){
+    py::class_<sycl::queue> syclqueue(m, "SyclQueue");
+    syclqueue.def(py::init<const sycl::device&>())
+        .def(py::init([](const std::string& filter) {
+                return get_queue_by_filter_string(filter);
+            })
+        )
+        .def(py::init([](const py::int_& obj) {
+                return get_queue_by_pylong_pointer(obj);
+            })
```

Comment:
Could you please share the case when it is needed? Does it covered by tests?

Reply:
https://github.com/pytorch/pytorch/blob/main/torch/csrc/xpu/Stream.cpp#L72

Reply:
Please add a comment somewhere that this is needed to accept pytorch tensors.
I think the better place for the comment is along with the function's definition. But it's up to you to decide about the comment's placement.

Reply:
I've now left a note @Vika-F please let me know what you think!

Reply:
I like the comment. It clarifies the things to me.